qemu/tests/set_link.py

```diff
@@ -77,48 +75,67 @@ def guest_reboot(reboot_method, link_up):
             if not link_up:
                 logging.info("Login error is expected when net link is down")
 
-    def guest_netwok_connecting_check(guest_ip, link_up, change_queues=False):
+    def ping_host_from_guest_check(session, link_up, guest_ifname=""):
         """
-        Check whether guest network is connective by ping
+        Check result about ping from guest to host
         """
+        timeout = 60
+        if params.get("os_type") == "linux":
+            ttl = "ttl="
+            command = "ping %s -c 10 -I %s" % (host_ip, guest_ifname)
+            if ":" in host_ip:
+                command = "ping6 %s -c 10 -I %s" % (host_ip, guest_ifname)
+        else:
+            ttl = "TTL="
+            command = "ping %s -n 10" % host_ip
         if link_up:
-            vm.wait_for_login(restart_network=True)
-            guest_ip = vm.get_address()
-        if change_queues:
-            env["run_change_queues"] = False
-            bg_thread = utils.InterruptedThread(change_queues_number_repeatly,
-                                                (guest_ifname,))
-            bg_thread.start()
-
-            utils_misc.wait_for(lambda: env["run_change_queues"], 30, 0, 2,
-                                "wait queues change start")
-
-        _, output = utils_test.ping(guest_ip, count=10, timeout=20)
-        if not link_up and utils_test.get_loss_ratio(output) != 100:
+            timeout = 360
+        end_time = time.time() + timeout
+        while time.time() < end_time:
+            output = session.cmd_output(command, timeout=60)
+            if ttl in output:
+                break
+            time.sleep(2)
+        logging.info("ping statistics:\n '%s'" % output)
+        if not link_up and ttl in output:
             err_msg = "guest network still connecting after down the link"
             raise error.TestFail(err_msg)
-        elif link_up and utils_test.get_loss_ratio(output) == 100:
+        elif link_up and ttl not in output:
```

Comment:
After set link off on windows, ping host will get "Destination net unreachable" and "0% loss". So use the ttl to check the connective of guest.